bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/BagsStatus.scala

```diff
@@ -0,0 +1,78 @@
+package edu.gemini.spModel.target.env
+
+import edu.gemini.shared.util.immutable.{Option => GOption}
+import edu.gemini.spModel.pio.{ParamSet, Pio, PioFactory}
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.rich.shared.immutable._
+
+// Possible BAGS statuses used by GuideProbeTargets to indicate the state of BAGS targets and lookups.
+sealed trait BagsStatus extends Cloneable {
+  val bagsStar: Option[SPTarget] = None
+  def bagsStarAsJava: GOption[SPTarget] = bagsStar.asGeminiOpt
+  override def clone: BagsStatus = this
```

Comment:
This looks a bit funny, is `BagsSuccessWithTarget` the special case? Could you use the new model `Target` instead of the old `SPTarget`?